src/main/scala/com/thenewmotion/chargenetwork/ocpp/charger/BosService.scala

```diff
@@ -134,25 +134,41 @@ class ConnectorServiceImpl(protected val service: SyncCentralSystem, connectorId
   }
 
   def signedMeterValuesBeginTransaction(meter: Int): Meter = {
-    val meterValue: Value = Value(value = meter.toString,
-      context = ReadingContext.TransactionBegin,
-      format = ValueFormat.Raw,
-      measurand = Measurand.EnergyActiveImportRegister,
-      phase = None,
-      location = Location.Outlet,
-      unit = UnitOfMeasure.Wh)
 
-    val signedMeterReading: Value = meterValue.copy(context = ReadingContext.TransactionBegin,
-      format = ValueFormat.SignedData, value = signedMeterValue)
+    val meterValue: Value = buildMeterValue(meter.toString, ReadingContext.TransactionBegin, ValueFormat.Raw)
+
+    val signedMeterReading: Value = meterValue.copy(format = ValueFormat.SignedData, value = signedMeterValueStart)
```

Comment:
Do you send the signed data always? i did not see a check for eichrecht stations

Reply:
This changes only affects Eichrecht. The decision is made on `signedMetersOnStop`. If it is an Eichrecht station it will send the meter values, otherwise it won't, as it would happen before these changes.

Reply:
No sure how this scala functions works but if someone calls this method and at line 142 is returning a meter object with the signed reading, it will be sent to everon ? If yes does it check somewhere if the station is eichrecht?

Reply:
Method `signedMeterValuesBeginTransaction` is used for abl only and its check is made on `startSession`.  In any case the meter value is sent because in Scala the last statement is always returned (unless it's a [void method](https://stackoverflow.com/questions/32018257/how-to-define-a-function-does-not-return-or-return-void-in-scala/32018603), which is not the case here)

The scope of this PR is on `stopSession`, in which start and end signed transaction data should be added to the transaction. In scala there is the concept of [nested method](https://docs.scala-lang.org/tour/nested-functions.html) in which you can create function inside functions and those exist only on the scope of the parent function. 

On `stopSessionMethod` there is this call: 

```scala
    service(StopTransactionReq(transactionId, card, ChargerClock.now, meterValue, stopReason, signedMetersOnStop))
      .idTag.exists(_.status == AuthorizationStatus.Accepted)
```

in which function `signedMetersOnStop` is used as parameter. 
```scala
def signedMetersOnStop: List[Meter] = {
      if (config.isEichrechtCharger) List(Meter(ChargerClock.now, getMeterValuesOnStop)) else Nil
    }
```

That's where the code decides if meter values should be sent or not. If it is Eichrecht it will send begin and end meter values. If not it will not send any meter value. 

`signedMetersOnStop` is called without `()` because it doesn't have any parameters. 

This is the stop transaction data sent to everon after those changes when a station is eichrecht (station id = EICHRECHT-P1739086): 
```
 "Incoming request: [2,"9","StopTransaction",{"transactionId":4,"idTag":"\"045F9CE24B4D80\"","timestamp":"2020-06-16T07:45:26.262786Z","meterStop":1070,"transactionData":[{"timestamp":"2020-06-16T07:45:26.264026Z",
"sampledValue[{ "value":"AP;0;3;ALCV3ABBBISHMA2RYEGAZE3HV5YQBQRQAEHAR2MN;BIHEIWSHAAA2W2V7OYYDCNQAAAFACRC2I4ADGAETI4AAAABAOOJYUAGMXEGV4AIAAEEAB7Y6AAO3EAIAAAAAAABQGQ2UINJZGZATGMJTGQ4DAAAAAAAAAACXAAAABKYAAAAA====;R7KGQ3CEYTZI6AWKPOA42MXJTGBW27EUE2E6X7J77J5WMQXPSOM3E27NMVM2D77DPTMO3YACIPTRI===;","context":"Transaction.End","format":"SignedData","unit":"kWh"},
{"value":"AP;1;3;ALCV3ABBBISHMA2RYEGAZE3HV5YQBQRQAEHAR2MN;BIHEIWSHAAA2W2V7OYYDCNQAAAFACRC2I4ADGAETI4AAAAAQHCKIUAETXIGV4AIAAEEAB7Y6ACT3EAIAAAAAAABQGQ2UINJZGZATGMJTGQ4DAAAAAAAAAACXAAAABLAAAAAA====;HIWX6JKGDO3JARYVAQYKJO6XB7HFLMLNUUCDBOTWJFFC7IY3VWZGN7UPIVA26TMTK4S2GVXJ3BD4S===;","context":"Transaction.Begin","format":"SignedData","unit":"kWh"}]}]}]" 
```

and this is the data sent to everon for any other station (station id = EVS-P17390866):
```
Incoming request: [2,"9","StopTransaction",{"transactionId":1,"idTag":"\"045F9CE24B4D80\"","timestamp":"2020-06-16T08:15:11.260078Z","meterStop":1010}]
```

Does it make sense?

Reply:
By the way, let's move this discussion to this PR, since this one is closed : https://github.com/akoshterek/ocpp-charger/pull/3